### PR TITLE
feat(sbb-tag): initial implementation

### LINKED ISSUE #1467: story(sbb-tag): initial implementation
### Feature Description

We want to provide a filter component. For this purpose, we want to implement `sbb-tag`.

[Figma Spec](https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?node-id=11020%3A397038&t=NU1HtEih6mLWGw6j-1)

**References**
- https://angular.app.sbb.ch/angular/components/tag/overview

**Design Spec**
- Native input type="checkbox" is used internally but visually hidden

**Technical Spec**
- Compare with implementation of `sbb-checkbox` and `sbb-tab-title`
- Forward `value` and `checked` property
- `sbb-tag` should have the `change` (and `@Event()` `didChange` for react support) event, which emits whenever selection changes
- Use the `label` element internally to enable any click on the `sbb-tag` to toggle checked status
- Provide `aria-label`, `aria-labelledby` and `aria-describedby` properties on `sbb-tag` and forward the content onto the native input element
- Implement a `sbb-tag-group` to group `sbb-tag` instances (compare with `sbb-checkbox-group`)
  - Should not consider nesting of groups (no `sbb-tag-group` inside another `sbb-tag-group`)
- For the amount, provide a named slot `amount` (The current `sbb-tab-amount` will most likely be refactored to a slot as well in the future)

```html
<sbb-tag name="example-name" value="example-value">
  Label
</sbb-tag>
<sbb-tag name="example-name2" value="example-value" checked>
  Checked Label
</sbb-tag>
```

Checkbox Group
```html
<sbb-tag-group>
  <sbb-tag name="example-name1" value="example-value 1">Label 1</sbb-tag>
  <sbb-tag name="example-name2" value="example-value 2">Label 2</sbb-tag>
  <sbb-tag name="example-name3" value="example-value 3">Label 3</sbb-tag>
</sbb-tag-group>
```

### Definition of Done

- [ ] Component is implemented
- [ ] Documentation is provided
- [ ] Tests are implemented
- [ ] E2E Tests are implemented
- [ ] Storybook stories are implemented
- [ ] Navigation via keyboard is tested
- [ ] Screen reader output is tested
- [ ] High-contrast is tested
- [ ] Remaining accessibility is tested
- [ ] UX approved

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #1467 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- sbb-tag implementation
- sbb-tag-group implementation

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [x] Edge Desktop
- [x] Safari Desktop
- [x] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [x] JAWS Firefox Desktop
- [x] JAWS Chrome Desktop
- [x] NVDA Firefox Desktop
- [x] NVDA Chrome Desktop
- [x] VoiceOver Safari Desktop
- [x] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-tag-group/sbb-tag-group.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+// Default component properties, defined for :host. Properties which can not
+// travel the shadow boundary are defined through this mixin
+@include sbb.host-component-properties;
+
+.sbb-tag-group {
+  display: flex;
+  flex-wrap: wrap;
+  gap: var(--sbb-spacing-fixed-3x);
+  margin-block-end: var(--sbb-spacing-responsive-m);
```
Comment (jeripeierSBB) [inline src/components/sbb-tag-group/sbb-tag-group.scss:11]: Manuel decided that this should not be a part of the component.
```suggestion
```

File: src/components/sbb-tag-group/sbb-tag-group.stories.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+const defaultArgTypes = {
+  checked,
+  disabled,
+  label,
+  value,
+  'icon-name': icon,
+  amount,
+  'accessibility-label': accessibilityLabel,
+  'accessibility-describedby': accessibilityDescribedby,
+  'accessibility-labelledby': accessibilityLabelledby,
```
Comment (jeripeierSBB) [inline src/components/sbb-tag-group/sbb-tag-group.stories.js:106]:
```suggestion
```

File: src/components/sbb-tag-group/sbb-tag-group.stories.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  decorators: [
+    (Story) => (
+      <div style={'padding: 2rem'}>
+        <Story />
+      </div>
+    ),
+  ],
+  parameters: {
+    actions: {
+      handles: ['change', events.didChange],
```
Comment (jeripeierSBB) [inline src/components/sbb-tag-group/sbb-tag-group.stories.js:187]:
```suggestion
      handles: ['change'],
```

File: src/components/sbb-tag-group/sbb-tag-group.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    if (isArrowKeyPressed(evt)) {
+      const cur: number = enabledTags.findIndex((e: HTMLSbbTagElement) => e === evt.target);
+      const nextIndex: number = getNextElementIndex(evt, cur, enabledTags.length);
+      enabledTags[nextIndex]?.focus();
+    }
+  }
+
+  public render(): JSX.Element {
+    if (this._isNested) {
+      throw new Error('Error: nested sbb-tag-group.');
```
Comment (jeripeierSBB) [inline src/components/sbb-tag-group/sbb-tag-group.tsx:56]: Is there a reason we need to catch this?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Partial review, only about sbb-tag-group. sbb-tag will follow tomorrow

File: src/components/sbb-tag-group/sbb-tag-group.tsx (hunk 4 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-tag-group/sbb-tag-group.tsx:56]: Taken from [here](https://github.com/lyne-design-system/lyne-components/issues/1467): "not consider nesting of groups (no sbb-tag-group inside another sbb-tag-group)".

Shoud I just return? or render some kind of error like [here](https://github.com/lyne-design-system/lyne-components/blob/master/src/components/sbb-card-product/sbb-card-product.tsx#L214)?

Comment (jeripeierSBB) [reply src/components/sbb-tag-group/sbb-tag-group.tsx:56]: I think we shouldn't do anything, it was just meant, that we don't have to support nested tag groups

**Pull request merged**